packages/backend/src/open_payments/auth/middleware.ts

```diff
@@ -80,3 +68,32 @@ export function createAuthMiddleware({
     }
   }
 }
+
+export const httpsigMiddleware = async (
+  ctx: HttpSigContext,
+  next: () => Promise<unknown>
+): Promise<void> => {
+  // TODO: look up client jwks.json
+  // https://github.com/interledger/rafiki/issues/737
+  if (!ctx.grant?.key.jwk) {
+    const logger = await ctx.container.use('logger')
+    logger.warn(
+      {
+        grant: ctx.grant
+      },
+      'missing grant key'
+    )
+    ctx.throw(500)
+  }
+  try {
+    if (
+      !(await validateSignature(ctx.grant.key.jwk, contextToRequestLike(ctx)))
+    ) {
+      ctx.throw(401, 'Invalid signature')
+    }
+  } catch (e) {
+    ctx.status = 401
+    ctx.throw(401, `Invalid signature`)
```

Comment:
```suggestion
    ctx.throw(401, `Invalid signature`)
```
is this redundant, or do we need to specify it before L92 as well?

Reply:
I think it's needed in case `validateSignature` throws.
I'll update to model after
https://github.com/interledger/rafiki/blob/683814d9dc71fd1d3b76f3e9aa251127e9af0c9e/packages/openapi/src/middleware.ts#L29-L30
(rethrow any koa `ctx.throw`s)

Reply:
I was thinking more so including `ctx.status = 401` _and_ also throwing with the same status. This makes sense though now